packages/react/src/menu/item/useMenuItem.ts

```diff
@@ -34,9 +33,9 @@ export function useMenuItem(params: useMenuItem.Parameters): useMenuItem.ReturnV
           id,
           role: 'menuitem',
           tabIndex: highlighted ? 0 : -1,
-          onKeyUp: (event: React.KeyboardEvent & MuiCancellableEvent) => {
+          onKeyUp: (event: BaseUIEvent<React.KeyboardEvent>) => {
             if (event.key === ' ' && typingRef.current) {
-              event.defaultMuiPrevented = true;
+              event.preventBaseUIHandler();
```

Comment:
cc @colmtuite you might want to mull over what the public API is for cancelling Base UI event handlers.

It's an advanced use case, but an unavoidable need if you are building a complex, highly interactive app

As an example from Radix, there you can do things like:
```jsx
onEscKeydown={event => event.preventDefault()}
```

It overloads the native `event.preventDefault()` to mean "prevent browser default AND the Radix handler", which may have been a bit of an overkill.

We relied on these escape hatches quite a lot back at Modulz when we needed more control over interactions in the product